linux/buildSrc/src/main/java/net/adoptopenjdk/installer/UploadRpmPackage.java

```diff
@@ -50,14 +59,37 @@ protected void uploadPackages(ArtifactoryCredentials artifactoryCredentials) {
                         getArchitecture()
                 );
 
-                executor.submit(UploadPackageTask.class, workerConfig -> {
-                    workerConfig.setIsolationMode(IsolationMode.NONE);
-                    workerConfig.setParams(getApiEndpoint(), artifactoryCredentials, upload);
-                });
+                this.getArtifactoryClient()
+                        .repository(upload.getRepository())
+                        .upload(upload.getRemotePath(), upload.getFile())
+                        .bySha1Checksum()
+                        .doUpload();
+            }
+        }
+    }
+
+    private boolean artifactExists(String remotePath) {
+        List<RepoPath> searchResults = this.getArtifactoryClient()
+                .searches()
+                .repositories(getRepository())
+                .artifactsByName(getPackageToPublish().getName())
+                .doSearch();
+
+        if (searchResults.isEmpty()) {
+            return false;
+        }
+
+        for (RepoPath searchResult : searchResults) {
+            if (Objects.equals(getRepository(), searchResult.getRepoKey())
+                    && Objects.equals(remotePath.substring(1), searchResult.getItemPath())) {
```

Comment:
slight preference to pull `remotePath.substring(1)` into a named variable,  but no biggy.